### PR TITLE
feat: Setup phpunit

### PR DESCRIPTION
**Pull request by provokateurin**
Depends on https://github.com/nextcloud/picker/pull/13

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,10 @@
 .vscode-upload.json
 .*.sw*
 /node_modules
+
+/vendor/
+<<<<<<< HEAD
```
Comment (bigcat88) [inline .gitignore:11]: what is this?

Comment (provokateurin) [reply .gitignore:11]: Me who was unable to fix the merge conflict :woman_facepalming:

**Comment by provokateurin** [comment]
@bigcat88 please review again :)

**Comment by bigcat88** [comment]
> @bigcat88 please review again :)

let's see this after the merge of #15

**Comment by provokateurin** [comment]
@bigcat88 it's ready now

**Comment by bigcat88** [comment]
is it intended  to have min `php 8.1`?

**Comment by provokateurin** [comment]
You mean 8.2 in the CI? That shouldn't matter as long as it is a version that is supported by server.

File: composer.lock [hunk 2]
```diff
@@ -144,152 +370,1539 @@
...
+                "reference": "355324ca4980b8916c18b9db29f3ef484078f26e",
                 "shasum": ""
             },
             "require": {
-                "php": ">=7.4.0"
+                "ext-dom": "*",
+                "ext-libxml": "*",
+                "ext-xmlwriter": "*",
+                "nikic/php-parser": "^4.15",
+                "php": ">=8.1",
```
Comment (bigcat88) [inline composer.lock:391]: I am about this, is this intended?

Comment (provokateurin) [reply composer.lock:391]: The lowest server version 24 already supports 8.1 so this is fine.

**Review by bigcat88 — APPROVED** [review]

**Pull request merged**